botocore/serialize.py

```diff
@@ -600,30 +609,41 @@ def _partition_parameters(
         location = member.serialization.get('location')
         key_name = member.serialization.get('name', param_name)
         if location == 'uri':
-            partitioned['uri_path_kwargs'][key_name] = param_value
-        elif location == 'querystring':
-            if isinstance(param_value, dict):
-                partitioned['query_string_kwargs'].update(param_value)
-            elif isinstance(param_value, bool):
+            if isinstance(param_value, bool):
                 bool_str = str(param_value).lower()
-                partitioned['query_string_kwargs'][key_name] = bool_str
+                partitioned['uri_path_kwargs'][key_name] = bool_str
             elif member.type_name == 'timestamp':
                 timestamp_format = member.serialization.get(
-                    'timestampFormat', self.QUERY_STRING_TIMESTAMP_FORMAT
+                    'timestampFormat', self.URI_TIMESTAMP_FORMAT
                 )
                 timestamp = self._convert_timestamp_to_str(
                     param_value, timestamp_format
                 )
-                partitioned['query_string_kwargs'][key_name] = timestamp
+                partitioned['uri_path_kwargs'][key_name] = timestamp
             else:
-                partitioned['query_string_kwargs'][key_name] = param_value
+                partitioned['uri_path_kwargs'][key_name] = param_value
+        elif location == 'querystring':
+            if isinstance(param_value, dict):
+                # Add only new query string key/value pairs.
+                # Named query parameters should take precedence.
+                for key, value in param_value.items():
+                    partitioned['query_string_kwargs'].setdefault(key, value)
```

Comment:
Isn't updating this to setdefault also breaking?  We are changing the value we send over the wire in some instances.

Have we looked at the actual methods that this change will impact to determine if it's safe to make?